arrow-data/src/ffi.rs

```diff
@@ -267,8 +267,19 @@ impl FFI_ArrowArray {
 
     /// the null count of the array
     #[inline]
-    pub fn null_count(&self) -> usize {
-        self.null_count as usize
+    pub fn null_count(&self) -> Option<usize> {
+        // MAY be -1 if not yet computed
+        if self.null_count >= 0 {
+            Some(self.null_count as usize)
+        } else {
+            None
+        }
```

Comment:
```suggestion
        self.null_count.try_into().ok()?
```
Perhaps, or using `bool::then_some` if you wish to be more explicit